core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -358,14 +358,12 @@ private[spark] class MesosClusterScheduler(
         .orElse(desc.command.environment.get("SPARK_EXECUTOR_URI"))
   }
 
-  private def getDriverEnvironment(desc: MesosDriverDescription): Environment = {
-    val env = {
-      val executorOpts = desc.conf.getAll.map { case (k, v) => s"-D$k=$v" }.mkString(" ")
-      val executorEnv = Map("SPARK_EXECUTOR_OPTS" -> executorOpts)
```

Comment:
From what I understand SPARK_EXECUTOR_OPTS is not used because it is actually reconstructed at the driver side
[here:](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala#L180)  from spark.executor.extraJavaOptions.
I dont see that property set at the Dispacther side.

Two things here:
I would prefer the process of reconstructing the env to be transparent to the user, the same for client and cluster mode, no need for special prefixes. 
Secondly, we need to make sure there is continuity with all properties we pass from process to process.